src/global/ValueId.h

```diff
@@ -345,7 +345,17 @@ class ValueId {
   /// and `ad_utility::HashMap`
   template <typename H>
   friend H AbslHashValue(H h, const ValueId& id) {
-    return H::combine(std::move(h), id._bits);
+    // Depending on the case a number is added to the hashing. This ensures that
+    // for two unequal elements the hash expansion (~list of simpler values
+    // being hashed with `combine`) of neither is a suffix of the other.
```

Comment:
I don't quite understand the comment,
Why is the added `0` for the bit hashing and `1` for the string hashing necessary?
Does it reduce the probability of collisions, or why do we need a suffix of hash expansions etc?

Reply:
Please elaborate and then we can reformulate.

Reply:
Okay, I don't see how `*id.getLocalVocabIndex()` can be equal to any of the `_bits` representations to make the added `0` or `1` necessary. Did you see test failures without them?

Reply:
Yes, I only added this because I saw that this condition was violated otherwise. This happened between an undefined ID  (hash expansion `[0]`) and any LocalVocabEntry Literal that is not in the original Index (hash expansion `["{Literal}", "{literal length as bytes}", 0]`). Without going into too much detail the `0` for the LVE Literal comes from it being index 0 in the LiteralOrIri variant.

Reply:
I guess it's hard to explain why this breaks in that specific case without going into the implementation details of absl. But it's easy to see that this ensures that it cannot happen without knowing much more. Maybe something likes this?
```suggestion
    // Adding 0/1 to the hash is required to ensure that for two unequal
    // elements the hash expansions of neither is a suffix of the other.This is
    // a property that absl requires for hashes. The hash expansion is the list
    // of simpler values actually being hashed (here: bits or hash expansion of
    // the LocalVocabEntry).
```